src/Model/User.php

```diff
@@ -466,19 +466,23 @@ public static function create(array $data)
 		// collapse multiple spaces in name
 		$username = preg_replace('/ +/', ' ', $username);
 
-		if (mb_strlen($username) > 48) {
-			throw new Exception(L10n::t('Please use a shorter name.'));
+		$username_min_length = max(1, min(255, intval(Config::get('system', 'username_min_length', 0))));
+		$username_max_length = max(1, min(255, intval(Config::get('system', 'username_max_length', 0))));
```

Comment:
We need some hard maximum limit. For example: The complete url of the contact must never exceed 255 characters, including the scheme. Otherwise we cannot store it any more in our database. Same is valid for the technical endpoints. They all contain the nick name. When one of the endpoints exceeds 255 characters, the communication will fail.

And you should possibly check if someone entered a minimum value that is higher than the maximum value.

Reply:
> We need some hard maximum limit.

That's exactly what `max(1, min(255, $value))` does, I'm glad you noticed it. I can lower it though.

> And you should possibly check if someone entered a minimum value that is higher than the maximum value.

What should have the priority, the minimum or the maximum?

Reply:
I ended up swapping the values if they are mutually exclusive and logging.

Reply:
The maximum length depends on so much things like the length of the domain or if Friendica is installed in a subfolder. However, I wouldn't like to make it much larger than before. So I would suggest taking some nice value larger than 48 - but not much larger.